paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -107,9 +130,74 @@ def paasta_mark_for_deployment(args):
     if service and service.startswith('services-'):
         service = service.split('services-', 1)[1]
     validate_service_name(service, soa_dir=args.soa_dir)
-    return mark_for_deployment(
+    ret = mark_for_deployment(
         git_url=args.git_url,
         deploy_group=deploy_group,
         service=service,
         commit=args.commit,
     )
+    if args.block:
+        try:
+            _log(
+                service=service,
+                line="Waiting for deployment of {0} to {1} complete".format(args.commit, deploy_group),
+                component='deploy',
+                level='event'
+            )
+            wait_for_deployment(args.service, args.deploy_group,
+                                args.commit, args.soa_dir, args.timeout)
+            _log(
+                service=service,
+                line="Deployment of {0} to {1} complete".format(args.commit, deploy_group),
+                component='deploy',
+                level='event'
+            )
+        except TimeoutError:
+            sys.exit(1)
+    return ret
+
+
+def is_instance_deployed(service, instance, git_sha):
+    api = client.get_paasta_api_client()
+    status = api.service.status_instance(service=service, instance=instance).result()
+    # if it's a chronos service etc then skip waiting for it to deploy
+    if not status.marathon:
+        return True
+    return git_sha.startswith(status.git_sha) and \
+        status.marathon.app_count == 1 and \
+        status.marathon.deploy_status == 'Running' and \
+        status.marathon.expected_instance_count == status.marathon.running_instance_count
+
+
+def wait_for_deployment(service, deploy_group, git_sha, soa_dir, timeout):
+    api = client.get_paasta_api_client()
+    instances = api.service.list_instances(service=service,
```

Comment:
This needs to be cluster-specific somehow right? because "main" may not be deployed in all the different clusters.